### PR TITLE
Add rustdoc performance blog post

### PR DESCRIPTION
**Pull request by jyn514**
r? @Manishearth 
cc @GuillaumeGomez

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
:+1:

**Comment by jyn514** [comment]
Oh hold on - @bugadani mentioned that we should probably mention the *actual* performance PRs, which are caching intra-doc links and not looking for blanket impls. Let's hold off on merging until then.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (therealprof) [inline posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:27]: That sounds a bit harsh and not environmentally friendly... maybe
```suggestion
### Pruning the tree
```

Comment (GuillaumeGomez) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:27]: Is it relevant here? It's a pun and it's funny.

Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:27]: How about a compromise: I change it to 'pruning the tree' if you let me add a footnote that it was changed to be environmentally friendly :joy: that just cracks me up.

**Comment by jyn514** [comment]
Ok, this is ready for review.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (bugadani) [inline posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:26]: - [#79682](https://github.com/rust-lang/rust/pull/77700)
+[#79682](https://github.com/rust-lang/rust/pull/79682)

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (Manishearth) [inline posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:115]:
```suggestion
## Why did we not rely more on rustc internals earlier?
```

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (Manishearth) [inline posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:127]: Readers may not know what doctree is

**Review by Manishearth — APPROVED** [review]
Looks good, might need someone to go through and make copy edits

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:127]: There's a section on doctree above: https://github.com/rust-lang/blog.rust-lang.org/pull/760/files#diff-894da88e61abfe9a88258302da89fc4eff778be7bcd7ca059081639b911733aeR51

Comment (GuillaumeGomez) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:127]: But before that it's put between backquotes. Maybe add backquotes on both `clean` and `doctree`? :)

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (GuillaumeGomez) [inline posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:170]: That makes me sad. It was a funny pun. :'(

**Review by camelid — COMMENTED** [review]
Great blog post! I have some comments that are mostly typographical nits and also a typo or two :)

Some of them are stylistic choices, so they're not a big deal, but I wanted to suggest possible alternatives to see what you think.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (GuillaumeGomez) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:10]: I used the twitter embed stuff, but just a link is fine I guess? For reference, here's the tweet URL: https://twitter.com/imperioworld_/status/1349383125051305984

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (GuillaumeGomez) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:41]: `stm32h7xx` should be between backquotes too.

Comment (camelid) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:41]: Great minds think alike ;)

https://github.com/rust-lang/blog.rust-lang.org/pull/760#discussion_r559036671

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (GuillaumeGomez) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:119]: rustc internals API doesn't get updated much anymore so we don't get breaking changes on rustdoc side like we used to.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (GuillaumeGomez) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:41]: :laughing:

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (camelid) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:10]: I'm just suggesting that we removed the `script` tag; the rest of it seems fine. We could use native Markdown, but I think we should at least remove the `script`.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (camelid) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:119]: It just seems redundant since the previous part of the sentence is saying that they're somewhat stable.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (camelid) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:24]: I think it's because it has a trailing period. Good catch!

```suggestion
[90000%]: https://github.com/rust-lang/rust/pull/77700#issuecomment-735995025
```

**Comment by jyn514** [comment]
@camelid I've given you write access to my fork, feel free to make all the edits you want. I'll try to address the review comments at some point but I may not get around to it today.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:152]: Oh oops - the one that's inconsistent is actually 'Removing doctree' - these are future plans and haven't been done yet, so 'ing' is wrong.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:142]: I don't think markdown can make separate paragraphs part of a list item even if they're indented. Instead I'll move the inlining bullet above the macros.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:41]: > It seems to me that the later paragraphs of each list item should be indented so they are recognized as part of the list item.

I don't think you can make later paragraphs part of the same list item.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:18]: I prefer it the way it is now, I think.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:119]: I removed it, I think the rest stands on its own.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:127]: Ok, I added backticks.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (camelid) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:142]: What do you mean? This works correctly:

```markdown
- Foo

  Bar
```

- Foo

  Bar

Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:142]: Oh huh, I didn't realize that worked. Thanks, I fixed it.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md]: Never mind, I fixed this.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (jyn514) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:10]: Without the script it does not look like a normal twitter embed:

image

So we kept it as-is.

File: posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md
Comment (bugadani) [inline posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:21]: Just realized, 90x =/= 90000%. I think this is a zero off :)

Comment (camelid) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:21]: The comment this links to seems to show an 82x speedup. Maybe we should just say "by a whopping 82x"?

Comment (bugadani) [reply posts/inside-rust/2021-01-15-rustdoc-performance-improvements.md:21]: a huge percentage looks  alot better tho 🤓

**Pull request merged**
